ggml.c

```diff
@@ -1523,6 +1523,214 @@ static inline void __sse_f16x4_store(ggml_fp16_t *x, __m128 y) {
 #define GGML_F16_VEC_MUL             GGML_F32Cx4_MUL
 #define GGML_F16_VEC_REDUCE          GGML_F32Cx4_REDUCE
 
+#elif defined(__loongarch_asx)
+
+#define GGML_SIMD
+
+// F32 LASX
+
+typedef union
+{
+    int32_t i;
+    float f;
+} FloatInt;
+/* float type data load instructions */
+static __m128 __lsx_vreplfr2vr_s(float val)
+{
+    FloatInt fi_tmpval = {.f = val};
+    return (__m128)__lsx_vreplgr2vr_w(fi_tmpval.i);
+}
+
+static __m256 __lasx_xvreplfr2vr_s(float val)
+{
+    FloatInt fi_tmpval = {.f = val};
+    return (__m256)__lasx_xvreplgr2vr_w(fi_tmpval.i);
+}
```

Comment:
Deduplicate this code by moving it in `ggml-impl.h` and reusing it in `ggml.c` and `ggml-quants.c`

Reply:
I was thinking to just deduplicate the `__lsx_vreplfr2vr_s` and `__lasx_xvreplfr2vr_s` code. The rest of the lsx/lasx code that is used only inside `ggml-quants.c` should remain in `ggml-quants.c`